cloud/vsphere/machineactuator.go

```diff
@@ -394,7 +400,8 @@ func (vc *VsphereClient) Delete(cluster *clusterv1.Cluster, machine *clusterv1.M
 
 	// The machine HCL reads the startup script, so we need to have something there for terraform
 	// to not fail.
-	if err := saveFile("", path.Join("/tmp", "machine-startup.sh"), 0644); err != nil {
+	startupScriptPath := path.Join("/tmp", startupScriptFilename)
```

Comment:
Why not just not delete the startup script until the machine is deleted?

Reply:
It is not just the contention with deletion.
If we have two workers creating two machine with different specs, there is a still contention about which startup script would be applied.

Reply:
Changed it to use /dev/null for deletes as there is no reason to create a temp dir/file if we simply need a place holder.